ethcore/engines/clique/src/lib.rs

```diff
@@ -250,17 +249,14 @@ impl Clique {
 		}
 	}
 
-	fn sign_header(&self, header: &Header) -> Result<(Signature, H256), Error> {
-
+	fn sign_header(&self, header: &Header) -> Result<(Signature, H256), EngineError> {
 		match self.signer.read().as_ref() {
-			None => {
-				Err(EngineError::RequiresSigner)?
-			}
+			None => Err(EngineError::RequiresSigner.into()),
 			Some(signer) => {
 				let digest = header.hash();
 				match signer.sign(digest) {
 					Ok(sig) => Ok((sig, digest)),
-					Err(e) => Err(EngineError::Custom(e.into()))?,
+					Err(e) => Err(From::from(EngineError::Custom(e.into()))),
```

Comment:
What is our policy for `From::from` vs `.into()`? I personally prefer the former, since it allows to explicitly specify a type, e.g.
`Error::from`. But mixing the two doesn't feel right.

Reply:
agree, I prefer explicitness too.

Reply:
I think the reason was that `from` didn't work for `parity_crypto::publickey::error::Error`